array_api_strict/tests/test_array_object.py

```diff
@@ -88,6 +88,14 @@ def test_validate_index():
     assert_raises(IndexError, lambda: a[0])
     assert_raises(IndexError, lambda: a[:])
 
+def test_promoted_scalar_inherits_device():
+    device1 = Device("device1")
+    x = asarray([1., 2, 3], device=device1)
+
+    y = x ** 2
```

Comment:
Without the change this line would fail, but I felt like having the `assert` below is somehow useful for readers from the future, even though it isn't strictly necessary